### PR TITLE
DDCE-265 validation to isaProviderRefNumber

### PR DESCRIPTION
**Pull request by prameya-phatak**

### CODE REVIEW

File: test/models/TradingDetailsSpec.scala [hunk 1]
```diff
@@ -25,14 +25,8 @@ class TradingDetailsSpec extends PlaySpec {
 
     "have no errors" when {
 
-      "both fields are correct - lowercase z in isa ref" in {
```
Comment (Ufuoma-Daniel) [inline test/models/TradingDetailsSpec.scala:28]: I'd have preferred us to keep this test but change the check to be that res.errors should not be Nil. That way we have confidence that lowercase z isn't being treated as valid. (This is me assuming there isn't already a test doing this).

Comment (prameya-phatak) [reply test/models/TradingDetailsSpec.scala:28]: added test case for invalid data i.e lowercase z

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
-
+!.gitignore
+.cache
```
Comment (Ufuoma-Daniel) [inline .gitignore:2]: Not sure why .cache was added, it should be --> !.github

File: test/models/TradingDetailsSpec.scala (hunk 1 above)
Comment (Ufuoma-Daniel) [reply test/models/TradingDetailsSpec.scala:28]: I think a test case needs to be added in this file too and not just at the controller level!

**Pull request merged**